app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -29,41 +30,64 @@
-    public static final String TITLE_RESID_INTENT_EXTRA = "extra_title_resid";
-
-    public static final int EXTRA_VALUE_NONE = -1;
+    public static final String SETTINGS_SCREEN_NAME = "settingsScreenName";
 
     private boolean localeUpdated;
+    private SettingsScreen settingsScreen;
 
     public interface ActionBarUpdater {
         void updateTitle(int titleResId);
         void updateIcon(int iconResId);
     }
 
-    public static SettingsFragment newInstance(Bundle intentArgs, int prefsResId, int titleResId) {
-        SettingsFragment f = new SettingsFragment();
-        f.setArguments(makeArgumentBundle(intentArgs, prefsResId, titleResId));
-        return f;
+    public enum SettingsScreen {
+        MAIN(R.xml.settings, R.string.menu_settings),
+        SEARCH_ENGINES(AppConstants.FLAG_MANUAL_SEARCH_ENGINE ?
+                R.xml.search_engine_settings_featureflag_manual :
+                R.xml.search_engine_settings,
+                R.string.preference_search_installed_search_engines),
+        ADD_SEARCH(R.xml.manual_add_search_engine, R.string.tutorial_search_title);
+
+        public final int prefsResId;
+        public final int titleResId;
+
+        SettingsScreen(int prefsResId, int titleResId) {
+            this.prefsResId = prefsResId;
+            this.titleResId =  titleResId;
+        }
     }
 
-    protected static Bundle makeArgumentBundle(Bundle intentArgs, int prefsResId, int titleResId) {
-        final Bundle args = intentArgs != null ? intentArgs : new Bundle();
-        args.putInt(PREFERENCES_RESID_INTENT_EXTRA, prefsResId);
-        args.putInt(TITLE_RESID_INTENT_EXTRA, titleResId);
-        return args;
+    public static SettingsFragment newInstance(Bundle intentArgs, SettingsScreen settingsType) {
+        final SettingsFragment f;
+        switch (settingsType) {
+            case MAIN:
+            case SEARCH_ENGINES:
+                f = new SettingsFragment();
+                break;
+            case ADD_SEARCH:
+                f = new ManualAddSearchEngineSettingsFragment();
+                break;
+            default:
+                f = new SettingsFragment();
+        }
+        if (intentArgs == null) {
+            intentArgs = new Bundle();
+        }
+        intentArgs.putString(SETTINGS_SCREEN_NAME, settingsType.name());
+        f.setArguments(intentArgs);
+        return f;
     }
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         final Bundle args = getArguments();
-        final int prefResId = args != null ?
-                args.getInt(PREFERENCES_RESID_INTENT_EXTRA, R.xml.settings) :
-                R.xml.settings;
+        settingsScreen = args != null ?
+                SettingsScreen.valueOf(args.getString(SETTINGS_SCREEN_NAME, SettingsScreen.MAIN.name())) :
+                SettingsScreen.MAIN;
```

Comment:
nit: When can args be null if we always set args? Add a comment.

I assume it's just because they're fragments and they're bullshit. x_x